geomstats/geometry/discrete_curves.py

```diff
@@ -1646,11 +1811,250 @@ def align(self, point, base_point):
     on the shape space of unparametrized curves, i.e. the space of parametrized
     curves quotiented by the group of reparametrizations. In the discrete case,
     reparametrization corresponds to resampling.
     """
+
+
+class SRVRotationBundle(FiberBundle):
+    """Principal bundle of curves modulo rotations with the SRV metric.
+
+    This is the fiber bundle where the total space is the space of parameterized
+    curves equipped with the SRV metric, the action is given by rotations, and
+    the base space is the shape space of curves modulo rotations.
+
+    Parameters
+    ----------
+    total_space : DiscreteCurvesStartingAtOrigin
+        Space of discrete curves starting at the origin
+    """
+
+    def __init__(self, total_space):
+        super().__init__(total_space=total_space)
+
+    @staticmethod
+    def _transpose(point):
+        """Transpose discrete curve starting at origin."""
+        return gs.moveaxis(point, -1, -2)
+
+    def _rotate(self, point, rotation):
+        """Rotate discrete curve starting at origin."""
+        return self._transpose(gs.matmul(rotation, self._transpose(point)))
+
+    def align(self, point, base_point, return_rotation=False):
+        """Align point to base point.
+
+        Find optimal rotation of curve with respect to base curve.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., k_sampling_points - 1, ambient_dim]
+            Discrete curve to align.
+        base_point : array-like, shape=[..., k_sampling_points - 1, ambient_dim]
+            Reference discrete curve.
+        return_rotation : boolean
+            If true, returns the optimal rotation used for the alignment.
+            Optional, default : False.
+
+        Returns
+        -------
+        aligned : array-like, shape=[..., k_sampling_points - 1, ambient_dim
+            Curve optimally rotated with respect to reference curve.
+        """
+        srv_transform = self.total_space.metric.diffeo
+        initial_srv = srv_transform.diffeomorphism(base_point)
+        end_srv = srv_transform.diffeomorphism(point)
+
+        mat = gs.matmul(self._transpose(initial_srv), end_srv)
+        u_svd, _, vt_svd = gs.linalg.svd(mat)
+        sign = gs.linalg.det(gs.matmul(u_svd, vt_svd))
+        vt_svd[..., -1, :] = gs.einsum("...,...j->...j", sign, vt_svd[..., -1, :])
+        rotation = gs.matmul(u_svd, vt_svd)
+
+        point_aligned = self._rotate(point, rotation)
+        if return_rotation:
+            return point_aligned, rotation
+
+        return point_aligned
+
+
+class SRVRotationQuotientMetric(QuotientMetric):
+    """SRV quotient metric on space of curves modulo rotations.
+
+    This is the class for the quotient metric induced by the SRV Metric on the
+    space of parametrized curves quotiented by the action of rotations.
+    """
+
+
+class SRVRotationReparametrizationBundle(SRVRotationBundle, SRVReparametrizationBundle):
```

Comment:
I think we should not inherit from these two, we should instantiate them (it avoids conflicts). i.e., in init:

```
self._rotations_bundle = SRVRotationBundle(total_space)
self._reparameterizations_bundle = SRVReparametrizationBundle(total_space)
```

and then we can use their methods properly by doing e.g. `self._rotations_bundle.align`.